### PR TITLE
pythonPackages.jupyter-server-proxy: init at 1.5.4

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

Needed for jupyterhub / jupyterlab deployments

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/jupyter-server-proxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    aiohttp
+  ];
+
+  checkInputs = [
+    pytest
+    pytest-asyncio
+  ];
+
+  # tests require network
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/jupyter-server-proxy/default.nix:36]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

File: pkgs/development/python-modules/jupyter-server-proxy/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  meta = with lib; {
+    description = "Jupyter server extension to supervise and proxy web services";
+    homepage = https://github.com/jupyterhub/jupyter-server-proxy;
```
Comment (jonringer) [inline pkgs/development/python-modules/jupyter-server-proxy/default.nix:40]: rfc 43
```suggestion
    homepage = "https://github.com/jupyterhub/jupyter-server-proxy";
```

File: pkgs/development/python-modules/simpervisor/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    aiohttp
+  ];
+
+  checkPhase = ''
+    pytest --ignore tests/test_ready.py
+  '';
+
+  meta = with lib; {
+    description = "Simple async process supervisor";
+    homepage = https://github.com/yuvipanda/simpervisor;
```
Comment (jonringer) [inline pkgs/development/python-modules/simpervisor/default.nix:32]:
```suggestion
    homepage = "https://github.com/yuvipanda/simpervisor";
```

File: pkgs/development/python-modules/jupyter-server-proxy/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    sha256 = "140mlzcnhadxrnhfwf7glx9cgzrcsbs1gmcsks62a8dpdns89spd";
+  };
+
+  propagatedBuildInputs = [
+    notebook
+    simpervisor
+    aiohttp
+  ];
+
+  checkInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/jupyter-server-proxy/default.nix:30]: We do not require checkInputs if no tests are run.

File: pkgs/development/python-modules/simpervisor/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "1brsisx7saf4ic0dih1n5y7rbdbwn1ywv9pl32bch3061r46prvv";
+  };
+
+  checkInputs = [
+    pytest
+    pytest-asyncio
+    aiohttp
+  ];
+
+  checkPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/simpervisor/default.nix:26]: Please use pytestCheckHook and disabledTests.

Comment (mweinelt) [reply pkgs/development/python-modules/simpervisor/default.nix:26]: `disabledTestFiles`, not `disabledTests`.

Comment (dotlambda) [reply pkgs/development/python-modules/simpervisor/default.nix:26]: And now actually `disabledTestPaths`.

**Pull request closed without merging**